test/object-store/sync/flx_schema_migration.cpp

```diff
@@ -660,8 +659,12 @@ TEST_CASE("An interrupted schema migration can recover on the next session",
     auto pf = util::make_promise_future<void>();
     config.sync_config->subscription_initializer = get_subscription_initializer_callback_for_schema_v1();
     config.sync_config->on_sync_client_event_hook = [&schema_version_changed_count, &task,
-                                                     &pf](std::weak_ptr<SyncSession>,
-                                                          const SyncClientHookData& data) mutable {
+                                                     &pf](std::weak_ptr<SyncSession> weak_session,
+                                                          const SyncClientHookData& data) {
+        auto session = weak_session.lock();
+        if (!session) {
+            return SyncClientHookAction::NoAction;
+        }
         if (data.event != SyncClientHookEvent::ErrorMessageReceived) {
             return SyncClientHookAction::NoAction;
         }
```

Comment:
Should there be a check on `task` for null before calling `task->cancel()` below
Although, it should be valid until the promise is completed.

Reply:
indeed. the task is reset after the promise is awaited.